compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1535,21 +1511,20 @@ pub trait PrettyPrinter<'tcx>:
                     }
                     _ => unreachable!(),
                 }
-
                 return Ok(self);
             }
-
-            (ConstValue::Scalar(scalar), _) => {
-                return self.pretty_print_const_scalar(scalar, ty, print_ty);
+            (ty::ValTree::Leaf(leaf), _) => {
+                return self.pretty_print_const_scalar_int(leaf, ty, print_ty);
             }
-
             // FIXME(oli-obk): also pretty print arrays and other aggregate constants by reading
             // their fields instead of just dumping the memory.
             _ => {}
         }
 
         // fallback
-        p!(write("{:?}", ct));
+        if valtree != ty::ValTree::zst() {
+            p!(write("{:?}", valtree));
+        }
```

Comment:
why that check? It looks to me like we would end up printing nothing for zsts then

Reply:
That was requested by @oli-obk.

Reply:
Zsts already print their type, printing their value has no ... value

Reply:
can we then also remove the `:` or something :sweat_smile: